newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/index.js

```diff
@@ -268,12 +268,48 @@ export default class ExpressionField extends React.Component<Props, State> {
     parameterValues: ParameterValues
   ) => {
     if (!this._inputElement) return;
+    const {
+      globalObjectsContainer,
+      objectsContainer,
+      expressionType,
+      value,
+    } = this.props;
     const cursorPosition = this._inputElement.selectionStart;
+    const parser = new gd.ExpressionParser2();
 
-    const functionCall = formatExpressionCall(expressionInfo, parameterValues);
+    // We add a character so that getCompletionDescriptionsFor will always return the type of the node
```

Comment:
I'm not familiar with this part of the code but this seems like a hack 🤔

Reply:
Indeed, @fannieyan can you explain why you need the type at a precise position?
it seems to be to check if some completions are strings, but why is this needed?

Reply:
This part is used when we choose an expression from the selector that displayed when clicking on the expression button (on the right of the field). It is then inserted where the caret was. There was previously no notion of type (except the type of the field) to determine what could be found in this selector.
It is the reason why I wanted to retrieve what type should the expression be here. I indeed hacked it so that the autocompletion finder is forced to return a type to me so that I would know what kind I needed/if I needed to convert the chosen expression. If a character is not added, if we insert an expression between parenthesis, it would not return autocompletion suggestions (it is as if we simulate that the user wants to type something).

Reply:
Can you try to use `ExpressionNodeLocationFinder::GetNodeAtPosition` to get the node at the cursor position, then `ExpressionTypeFinder::GetType` to get the type of this node?

I think this would be less hacky than adding a character and then looking at the completions (there could be none for a good reason).

You'll need to expose these two static methods from the C++, let me know if you struggle to do that (search for `STATIC_` in Bindings.idl)